src/shell/lisa_shell

```diff
@@ -318,14 +327,22 @@ EOF
 }
 
 function _lisa-test {
-nosetests -v --nocapture --nologcapture \
-          --logging-config=logging.conf \
-          $*
+if [ -n "$LISA_RESULTS_DIR" ]; then
```

Comment:
Can we add at the top a default initialization for the results dir?
Something like:
```bash
LISA_RESULTS_DIR=${LISA_RESULTS_DIR:-'lisa_tests_results'}
```
not sure if the default makes sense... to be verified.

This will allow to skip this `if` and use instead

```bash
[ -d "$LISA_RESULTS_DIR" ] || mkdir -p "$LISA_RESULTS_DIR"
local xunit_options=(--with-xunit --xunit-file="$LISA_RESULTS_DIR/results.xml")
```

Reply:
That is unfortunately not possible. This directory is by default based on the date, so no fixed default is acceptable. If there is no fixed default, there is no way for env.py to know if this variable is set by the user of set as a default value, unless you add an extra variable LISA_RESULTS_DIR_IS_DEFAULT=0/1  to signal that. If you don't have this information, the Python code can't decide whether to use LISA_RESULTS_DIR or target.config settings.

Reply:
cannot we have the current data expanded as a default in the above initialisation?

I mean... something like:
```bash
LISA_RESULTS_DIR=${LISA_RESULTS_DIR:-$(date +%Y%m%d)'}
```
where you use the format string used somewhere else to initialize that folder?

Reply:
If you do that, what behavior do you want when you set "results-dir" in target.config ? The only solution would be to have target.config winning over the environment variable. This is broken since the XML file would end up in the wrong directory in that case. Going that way means also removing the option from target.config ...

Reply:
+1